CodeEdit/Features/NavigatorSidebar/SourceControlNavigator/SourceControlToolbarBottom.swift

```diff
@@ -23,14 +53,35 @@ struct SourceControlToolbarBottom: View {
 
     private var sourceControlMenu: some View {
         Menu {
-            Button("Discard Changes...") {}
-                .disabled(true) // TODO: Implementation Needed
-            Button("Stash Changes...") {}
-                .disabled(true) // TODO: Implementation Needed
-            Button("Commit...") {}
-                .disabled(true) // TODO: Implementation Needed
-            Button("Create Pull Request...") {}
-                .disabled(true) // TODO: Implementation Needed
+            Button("Discard Changes") {
+                shell("cd \(getCurrentWorkspaceDocument(workspace: workspace)); git reset –hard")
+            }
+            Button("Stash Changes") {
+                shell("cd \(getCurrentWorkspaceDocument(workspace: workspace)); git stash")
+            }
+            Button("Commit") {
+                // TODO: Handle output
+                var file = getCurrentWorkspaceDocument(workspace: workspace)
+                if !commitText.isEmpty {
+                    commited = true
+                    shell("cd \(file); git add .")
+                    shell("cd \(file); git commit -m '\(commitText)'")
```

Comment:
This will fail if someone uses ' in their commit message, you should escape this to prevent errors.

Reply:
Good catch, @0xWDG! I'll fix that.

Reply:
Fixed in the latest commit :)